lock-impl/src/test/java/com/palantir/lock/impl/AdjustableBackgroundTaskTest.java

```diff
@@ -0,0 +1,135 @@
+    private final AtomicInteger numCallsToShouldRunSupplier = new AtomicInteger();
+    private final AtomicInteger numCallsToIntervalSupplier = new AtomicInteger(0);
+    private final AdjustableBackgroundTask adjustableBackgroundTask = new AdjustableBackgroundTask(
+            () -> {
+                numCallsToShouldRunSupplier.incrementAndGet();
+                return shouldRun.current();
+            },
+            () -> {
+                numCallsToIntervalSupplier.incrementAndGet();
+                return interval.current();
+            },
+            field::incrementAndGet,
+            scheduledExecutor);
+
+    @BeforeClass
+    public static void testDelayShouldNotExceedMinimumDelay() {
+        // We have to ensure that our default interval is higher than MINIMUM_INTERVAL_IF_NOT_RUNNING
+        // so our tests are actually meaningful
+        assertThat(DEFAULT_INTERVAL).isGreaterThanOrEqualTo(AdjustableBackgroundTask.MINIMUM_INTERVAL_IF_NOT_RUNNING);
+    }
+
+    @Test
+    public void doesNotRunTaskDuringConstruction() {
+        assertThat(field.get()).isEqualTo(0);
+    }
+
+    @Test
+    public void doesNotRunTaskByDefault() {
+        tick(DEFAULT_INTERVAL);
+        assertThat(field.get()).isEqualTo(0);
+    }
+
+    @Test
+    public void canBeEnabledDisabledAndReEnabled() {
+        shouldRun.update(true);
+        tick(DEFAULT_INTERVAL);
+        int after = field.get();
+        assertThat(after).isEqualTo(1);
+
+        shouldRun.update(false);
+        tick(DEFAULT_INTERVAL);
+        assertThat(field.get()).isEqualTo(after);
+
+        shouldRun.update(true);
+        tick(DEFAULT_INTERVAL);
+        assertThat(field.get()).isEqualTo(after + 1);
+    }
+
+    @Test
+    public void canAdjustInterval() {
+        Duration newInterval = Duration.ofMinutes(1);
+        interval.update(newInterval);
+        shouldRun.update(true);
+        // will run the task once
+        tick(DEFAULT_INTERVAL);
+        tick(Duration.ofMinutes(42 - 1));
+        assertThat(field.get()).isEqualTo(42);
+    }
+
+    @Test
+    public void canAdjustIntervalBelowMinimumDelayIfRunning() {
+        Duration newInterval = Duration.ofMillis(1);
+        interval.update(newInterval);
+        shouldRun.update(true);
+        // will run the task once
+        tick(DEFAULT_INTERVAL);
+        tick(Duration.ofMillis(10 - 1));
+        assertThat(field.get()).isEqualTo(10);
+    }
+
+    @Test
+    public void runsSuppliersAtReducedIntervalIfNotRunning() {
+        Duration newInterval = Duration.ofMillis(1);
+        interval.update(newInterval);
+        Duration elapsedDuration = Duration.ofSeconds(10);
+        // will invoke suppliers an additional time
+        tick(DEFAULT_INTERVAL);
+        tick(elapsedDuration);
+        assertThat(field.get()).isEqualTo(0);
+        // suppliers are also called once when running the constructor
+        assertThat(numCallsToShouldRunSupplier.get())
```

Comment:
nit: maybe use the method `as` here to explain a bit more clearly what you're testing. I.e. `Ensure the number of times we checked to run is at most the number of times the MINIMUM_INTERVAL_IF_NOT_RUNNING has elapsed.`